craft_application/models/project.py

```diff
@@ -167,8 +167,10 @@ def from_platforms(cls, platforms: craft_platforms.Platforms) -> dict[str, Self]
         return result
 
 
-def _populate_platforms(platforms: dict[str, Any]) -> dict[str, Any]:
-    """Populate empty platform entries.
+def _expand_shorthand_platforms(platforms: dict[str, Any]) -> dict[str, Any]:
+    """Expand shorthand platform entries into standard form.
+
+    Assumes the platform label is a valid architecture.
```

Comment:
Or base:architecture.

Reply:
Good call, since using this function is risky for multi-base platforms, I reworded to make it clear but not prescriptive.